internal/app/store/kvstore/store.go

```diff
@@ -0,0 +1,138 @@
+package kvstore
+
+import (
+	"errors"
+	"github.com/gasparian/money-transfers-api/internal/app/models"
+	"sort"
+	"sync"
+	"time"
+)
+
+var (
+	accNotFoundErr        = errors.New("Account not found")
+	notEnoghMoneyOnAccErr = errors.New("There is no enough money on account to complete a transaction")
+)
+
+type ConcurrentAccount struct {
+	models.Account
+	mx sync.RWMutex
+}
+
+type KVStore struct {
+	mx               sync.RWMutex
+	accIncID         int64
+	transactionIncID int64
+	accounts         map[int64]*ConcurrentAccount
+	transactions     map[int64]models.Transaction
+}
+
+func New() *KVStore {
+	return &KVStore{
+		accounts:     make(map[int64]*ConcurrentAccount),
+		transactions: make(map[int64]models.Transaction),
+	}
+}
+
+func (s *KVStore) InsertAccount(balance models.MoneyAmount) (models.Account, error) {
+	s.mx.Lock()
+	defer s.mx.Unlock()
+
+	s.accIncID++
+	acc := &ConcurrentAccount{
+		Account: models.Account{
+			AccountID: s.accIncID,
+			CreatedAt: time.Now(),
+			Balance:   balance,
+		},
+	}
+	s.accounts[s.accIncID] = acc
+	return acc.Account, nil
+}
+
+func (s *KVStore) DeleteAccount(accId int64) error {
```

Comment:
Will this class work correctly if I call this sequence of operations on an empty KVStore:
1. DeleteAccount(1)
2. DeleteAccount(2
3. AddAccount(200) -- Here we should fail because s.accIncID will be negative

Reply:
oh, my bad. Maybe it's just better to always guarantee that accID won't be negative. And give an error in case of deleting the non-existent account.